text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,730 @@
+
+### English
+[english]: #english
+
+This section applies to `rustc` and the standard library.
+
+An additional suggestion over RFC 505: One specific rule that comes up often:
+when quoting something for emphasis, use a single quote, and put punctuation
+outside the quotes, ‘this’. When quoting something at length, “use double
+quotes and put the punctuation inside of the quote.” Most documentation will
+end up using single quotes, so if you’re not sure, just stick with them.
+
+### Using Markdown
+[using-markdown]: #using-markdown
+
+The updated list of common headings is:
+
+* Examples
+* Panics
+* Errors
+* Safety
+* Aborts
+* Undefined Behavior
+
+RFC 505 suggests that one should always use the `rust` formatting directive:
+
+    ```rust
+    println!("Hello, world!");
+    ```
+
+    ```ruby
+    puts "Hello"
+    ```
+
+But, in API documentation, feel free to rely on the default being ‘rust’:
+
+    /// For example:
+    ///
+    /// ```
+    /// let x = 5;
+    /// ```
+
+Other places do not know how to highlight this anyway, so it's not important to
+be explicit.
+
+RFC 505 suggests that references and citation should be linked ‘reference
+style.’ This is still recommended, but prefer to leave off the second `[]`:
+
+```
+[Rust website]
+
+[Rust website]: http://www.rust-lang.org
+```
+
+to
+
+```
+[Rust website][website]
+
+[website]: http://www.rust-lang.org
+```
+
+But, if the text is very long, it is okay to use this form.
+
+### Examples in API docs
+[examples-in-api-docs]: #examples-in-api-docs
+
+Everything should have examples. Here is an example of how to do examples:
+
+```
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, "bar");
+/// ```
+///
+/// Passing a closure to compare with, rather than a string:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, |a| a.chars().is_whitespace().all());
+/// ```
+```
+
+For particularly simple APIs, still say “Examples” and “Basic usage:” for
+consistency’s sake.
+
+### Referring to types
+[referring-to-types]: #referring-to-types
+
+When talking about a type, use its full name. In other words, if the type is generic,
+say `Option<T>`, not `Option`. An exception to this is bounds. Write `Cow<'a, B>`
+rather than `Cow<'a, B> where B: 'a + ToOwned + ?Sized`.
+
+Another possibility is to write in lower case using a more generic term. In other words,
+‘string’ can refer to a `String` or an `&str`, and ‘an option’ can be ‘an `Option<T>`’.
+
+### Use parentheses for functions
+[use-parentheses-for-functions]: #use-parentheses-for-functions
+
+When referring to function names, include the `()`s after the name. For example, do this:
+
+```rust
+/// This behavior is similar to the way that `mem::replace()` works.
+```
+
+Not this:
+
+```rust
+/// This behavior is similar to the way that `mem::replace` works.
+```
+
+This helps visually differentiate it in the text.
```

Comment:
To address this reason, there is already plenty of visual distinction in both the Markdown, with the backticks and if the other guidelines are followed the link brackets, and in the rendered form, in the monospace font, grey background, and if other guidelines are followed, the link highlighting. I don't know if the additional visual differentiation of the parens is necessary; the only lowercase item in another module you could be referring to is a function, I believe, unless someone violates naming conventions.